search-ui/src/components/FilingHistory/FilingHistory.vue

```diff
@@ -0,0 +1,617 @@
+              </template>
+
+              <!-- the documents section -->
+              <template v-if="filing.documents && filing.documents.length > 0">
+                <v-divider class="my-6" />
+                <DocumentsList :filing=filing :loadingOne=loadingOne :loadingAll=loadingAll
+                  :loadingOneIndex=loadingOneIndex @downloadOne="downloadOne" @downloadAll="downloadAll($event)"
+                  :isLocked=isLocked />
+              </template>
+
+              <!-- the details (comments) section -->
+              <template v-if="filing.comments && filing.commentsCount > 0">
+                <v-divider class="my-6" />
+                <DetailsList :filing=filing :isTask="false" />
+              </template>
+
+            </div>
+          </v-expansion-panel-text>
+        </v-expansion-panel>
+      </v-expansion-panels>
+      <!-- No Results Message -->
+      <v-card class="no-results" flat v-else>
+        <v-card-text>
+          <div class="no-results__title">You have no filing history</div>
+          <div class="no-results__subtitle">Your completed filings and transactions will appear here</div>
+        </v-card-text>
+      </v-card>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+
+// Components and dialogs
+import CompletedAlteration from './CompletedAlteration.vue'
+import CompletedDissolution from './CompletedDissolution.vue'
+import DocumentsList from './DocumentsList.vue'
+import FiledLabel from './FiledLabel.vue'
+import FutureEffective from './FutureEffective.vue'
+import FutureEffectivePending from './FutureEffectivePending.vue'
+import PaperFiling from './PaperFiling.vue'
+import PendingFiling from './PendingFiling.vue'
+import StaffFiling from './StaffFiling.vue'
+import { DetailsList } from '@/components/common'
+import {
+  isStatusPaid, isStatusCompleted, isTypeAlteration, isTypeChangeOfAddress,
+  isEffectOfOrderPlanOfArrangement, isTypeDissolution, isTypeStaff, filingTypeToName, camelCaseToWords,
+  dateToYyyyMmDd, dateToPacificDateTime, flattenAndSortComments, isEffectiveDateFuture, isEffectiveDatePast
+} from '@/utils'
+import { fetchDocument, fetchDocuments, fetchComments } from '@/requests'
+import { LegalFiling, ApiFiling } from '@/interfaces/legal-api-responses'
+
+// Enums, interfaces and mixins
+import { ref, computed, watch } from 'vue'
+import { ErrorCategories, ErrorCodes, FilingTypes } from '@/enums'
+import { Document, FilingHistoryItem } from '@/types'
+import { useEntity, useFilingHistory } from '@/composables'
+import { ErrorI } from '@/interfaces'
+import { StatusCodes } from 'http-status-codes'
+
+const { entity } = useEntity()
+const { filingHistory } = useFilingHistory()
+
+const isBComp = computed(() => (entity.legalType == 'BEN') as boolean)
+const identifier = computed(() => entity.identifier as string)
+const legalName = computed(() => entity.name as string)
+const legalType = computed(() => entity.legalType as string)
```

Comment:
I think you should remove the casting (as ____) here because it will pickup the typing without it (i.e. legalType will be of type BusinessStatuses) and that way if it changes in the entity-factory it will pick up the warning here

Reply:
Ya will move this to composable